include/multipass/top_catch_all.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2020 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_TOP_CATCH_ALL_H
+#define MULTIPASS_TOP_CATCH_ALL_H
+
+#include <multipass/format.h>
+#include <multipass/logging/log.h>
+
+namespace multipass
+{
+template <typename F, typename... Args>                                      // F needs to return int
```

Comment:
Altough the function is clear, I'd add a clear comment on what its purpose is. It's just a suggestion, I generally try to comment everything not trivial.

Reply:
Sure thing, I will do that!

Reply:
Done!